crates/sui-storage/src/mutex_table.rs

```diff
@@ -103,6 +103,8 @@ impl<K: Hash + Eq + Send + Sync + 'static, L: Lock + 'static> LockTable<K, L> {
         cleanup_initial_delay: Duration,
         cleanup_entries_threshold: usize,
     ) -> Self {
+        let num_shards = if cfg!(msim) { 4 } else { num_shards };
```

Comment:
why is it a new problem though? 
AuthorityStore already had mutex_table field which used the same parameters. Or doubling pushed it over the edge?

Reply:
Its not a new problem at, but I noticed it because the nightly simulator run is OOMing (we run many tests in parallel). I'm also not convinced this isn't a problem on the validators. The MutexTable is currently consuming hundreds of megs of ram which seems crazy to me.